### PR TITLE
chore: remove unnecessary `list_files` during build copy-into physical pipeline

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- remove unnecessary `list_files` during build copy-into physical pipeline
- new metrics (to be detailed)

- `CopyIntoTable::files` has been removed, since nobody used it

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14819)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_copy_into_table.rs [hunk 1]
```diff
@@ -120,7 +122,15 @@ impl CopyIntoTableInterpreter {
                 &plan.table_name,
             )
             .await?;
-        let files = plan.collect_files(self.ctx.as_ref()).await?;
+        eprintln!(
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_copy_into_table.rs:125]: .

File: src/query/storages/fuse/src/operations/common/snapshot_generator.rs [hunk 2]
```diff
@@ -456,6 +456,10 @@ impl SnapshotGenerator for AppendGenerator {
             .ctx
             .get_settings()
             .get_auto_compaction_imperfect_blocks_threshold()?;
+        eprintln!(
+            "imperfect_count {}, auto_compaction_imperfect_blocks_threshold {}",
```
Comment (sundy-li) [inline src/query/storages/fuse/src/operations/common/snapshot_generator.rs:460]: debug msgs

Comment (dantengsky) [reply src/query/storages/fuse/src/operations/common/snapshot_generator.rs:460]: cleaning up

**Comment by dantengsky** [comment]
replace by the following PRs
- #15005  
- #14962

**Pull request closed without merging**